xbmc/pictures/ExifParse.cpp

```diff
@@ -376,7 +377,7 @@ void CExifParse::ProcessDir(const unsigned char* const DirStart,
       unsigned OffsetVal;
       OffsetVal = (unsigned)Get32(DirEntry+8, m_MotorolaOrder);
       // If its bigger than 4 bytes, the dir entry contains an offset.
-      if (OffsetVal+ByteCount > ExifLength)
+      if (OffsetVal > UINT32_MAX - ByteCount || OffsetVal + ByteCount > ExifLength)
```

Comment:
```suggestion
      if (OffsetVal > std::numeric_limits<uint32_t>::max() - ByteCount || OffsetVal + ByteCount > ExifLength)
```

Reply:
As long as we have an "upstream" copy - I would suggest to stay as near to it as possible to ease future cherry-pick / fixes.

Please merge ahead if you agree, seeing that you already acked.